File: backend/src/main/java/org/cwowhappy/securityanalyze/stock/domain/repository/StockRepository.java
package org.cwowhappy.securityanalyze.stock.domain.repository;

import org.cwowhappy.securityanalyze.shared.dto.PageQuery;
import org.cwowhappy.securityanalyze.shared.dto.PageResult;
import org.cwowhappy.securityanalyze.stock.domain.model.Stock;
import org.cwowhappy.securityanalyze.stock.domain.model.StockId;

import java.util.List;
import java.util.Optional;

/**
 * 股票仓库接口（Port）。
 * 定义在领域层，由基础设施层实现。
 */
public interface StockRepository {

    Optional<Stock> findById(StockId id);

    Optional<Stock> findByStockCode(String stockCode);

    List<Stock> findAll();

    PageResult<Stock> findByPage(PageQuery query, String market, String industry, String area, String keyword);

    List<Stock> findByCompanyId(String companyId);

    /**
     * 按行业查询股票列表。
     */
    List<Stock> findByIndustry(String industry);

    StockId save(Stock stock);

    void deleteById(StockId id);
}

File: collector/src/stock_collector/core/ports/stock_repository.py
"""股票仓库接口（Port）。"""

from abc import ABC, abstractmethod
from collections.abc import Sequence

from stock_collector.core.domain.stock import Stock


class StockRepository(ABC):
    """股票仓库抽象接口。"""

    @abstractmethod
    def save(self, stock: Stock) -> None:
        """保存股票数据。"""

    @abstractmethod
    def find_by_symbol(self, symbol: str) -> Stock | None:
        """根据股票代码查询。"""

    @abstractmethod
    def find_all(self) -> Sequence[Stock]:
        """查询所有股票。"""
